draft best practices for production workloads (#103)
GameFabric/docs: src/multiplayer-servers/production-workloads/requirements.md

    * **Ready State**: Signaling when the server is ready to accept players.
    * **Allocated State**: Signaling when the server is in use with active players. Any other state might get terminated by the system if necessary.
    * **Health Checks**: Periodically pinging to indicate the server is alive and healthy.

See [Using the Agones SDK](/multiplayer-servers/getting-started/using-the-agones-sdk) for implementation details.

### Lifecycle management

#### Armadas

After a session ends (for example, when the last player leaves), the game server should either return to an `Agones Ready` state if it’s meant to be reused for new sessions, or shut down if it’s no longer needed; in addition, the maximum lifetime of any game server should be limited to 24 hours.

#### Formations

For Formations, you must provide shutdown hints (required for Formations) as described in [running your game server](/multiplayer-servers/getting-started/running-your-game-server#termination-grace-periods).

Refer to [Vessel Shutdown Behavior](/multiplayer-servers/getting-started/terminating-game-servers#vessels) for additional details.

## Best Practices

> * **Keep your game server image as small as possible**: A small game server image makes everything faster and will also save money in the long run.

Reviewer: ```suggestion
* **Keep your game server image as small as possible**: A small game server image makes everything faster and also saves money in the long run.
```

* **Log to stdout**: Whenever possible, configure your game server to write logs directly to the standard output (`stdout`) and standard error (`stderr`) streams. This allows for seamless integration with GameFabric's log aggregation and monitoring systems. If your server can only write to a file, use the `gswrapper

## Game Server Wrapper (gswrapper)

The `gswrapper` is a valuable tool for launching your game server executable within the GameFabric environment. It is production-safe and offers several features to simplify integration and improve reliability.

### Key Features:

* **Parameter and Configuration Templating**: Dynamically configure your game server's command-line arguments and configuration files with runtime information from Agones, such as IP addresses and ports.
* **Log Tailing**: The wrapper can tail your game server's log files, which is useful if your server writes logs to a file instead of `stdout`.
* **Post-Stop Hooks (Advanced Crash Handling)**: You can configure executables to run after your game server stops. This is useful for tasks like analyzing core dumps for crash reporting or uploading logs for further investigation.

> For more information, you can refer to the [Game Server Wrapper documentation](/multiplayer-servers/multiplayer-services/game-server-wrapper) and the [gswrapper GitHub repository](https://github.com/GameFabric/gswrapper).

Reviewer: ```suggestion
For more information, please refer to the [Game Server Wrapper documentation](/multiplayer-servers/multiplayer-services/game-server-wrapper) and the [gswrapper GitHub repository](https://github.com/GameFabric/gswrapper).
```

    * **Ready State**: Signaling when the server is ready to accept players.
    * **Allocated State**: Signaling when the server is in use with active players. Any other state might get terminated by the system if necessary.
    * **Health Checks**: Periodically pinging to indicate the server is alive and healthy.

See [Using the Agones SDK](/multiplayer-servers/getting-started/using-the-agones-sdk) for implementation details.

### **Lifecycle management**:

#### Armadas

* After a session ends (e.g. last player leaves), either:
    * Return to `Agones Ready` if the server should be reused for new sessions, or
    * Shutdown if the server is no longer needed
* limit the max game server lifetime to 24h

#### Formations

* **Shutdown hints**: (Required for Formations) see [running your game server](/multiplayer-servers/getting-started/running-your-game-server#termination-grace-periods).
* See [Vessel Shutdown Behavior](/multiplayer-servers/getting-started/terminating-game-servers#vessels).

## Best Practices

Following these best practices will help ensure your game servers run smoothly and efficiently on GameFabric.

> * **Log to stdout**: Whenever possible, configure your game server to write logs directly to the standard output (`stdout`) and standard error (`stderr`) streams. This allows for seamless integration with GameFabric's log aggregation and monitoring systems. If your server can only write to a file, use the `gswrapper` tool to tail your log files and forward them to stdout/stderr. See the section below for more details.

Reviewer: In the `Best Practices` section we have only a single bullet point? seems like the section is unnecessary 😄

## Game Server Wrapper (gswrapper)

The `gswrapper` is a valuable tool for launching your game server executable within the GameFabric environment. It is production-safe and offers several features to simplify integration and improve reliability.

### Key Features

* **Parameter and Configuration Templating**: Dynamically configure your game server's command-line arguments and configuration files with runtime information from Agones, such as IP addresses and ports.
* **Log Tailing**: The wrapper can tail your game server's log files, which is useful if your server writes logs to a file instead of `stdout`.
* **Post-Stop Hooks (Advanced Crash Handling)**: You can configure executables to run after your game server stops. This is useful for tasks like analyzing core dumps for crash reporting or uploading logs for further investigation.

> For more information, please refer to the [Game Server Wrapper documentation](/multiplayer-servers/multiplayer-services/game-server-wrapper) and the [gswrapper GitHub repository](https://github.com/GameFabric/gswrapper).

Reviewer: How is this relevant as part of the `requirements.md` file?

Author: thanks
